src/authTokenInterceptor.ts

```diff
@@ -182,33 +148,30 @@ export const authTokenInterceptor = ({
   return async (requestConfig: any): Promise<any> => {
     // Waiting for a fix in axios types
     // We need refresh token to do any authenticated requests
-    if (!getRefreshToken()) return requestConfig
-
-    // Queue the request if another refresh request is currently happening
-    if (isRefreshing) {
-      return new Promise((resolve, reject) => {
-        queue.push({ resolve, reject })
-      })
-        .then((token) => {
-          if (requestConfig.headers) {
-            requestConfig.headers[header] = `${headerPrefix}${token}`
-          }
-          return requestConfig
-        })
-        .catch(Promise.reject)
-    }
-
-    // Do refresh if needed
-    let accessToken
-    try {
-      accessToken = await refreshTokenIfNeeded(requestRefresh)
-      resolveQueue(accessToken)
-    } catch (error: unknown) {
-      if (error instanceof Error) {
-        declineQueue(error)
-        throw new Error(
-          `Unable to refresh access token for request due to token refresh error: ${error.message}`
-        )
+    if (!(await getRefreshToken())) return requestConfig
+
+    let accessToken = undefined
+
+    // Try to await a current request
+    if (currentlyRequestingPromise) accessToken = await currentlyRequestingPromise
+
+    if (!accessToken) {
+      try {
+        // Sets the promise so everyone else will wait - then get the value
+        currentlyRequestingPromise = refreshTokenIfNeeded(requestRefresh)
+        accessToken = await currentlyRequestingPromise
+
+        // Reset the promise
+        currentlyRequestingPromise = undefined
+      } catch (error: unknown) {
+        // Reset the promise
+        currentlyRequestingPromise = undefined
+
+        if (error instanceof Error) {
+          throw new Error(
+            `Unable to refresh access token for request due to token refresh error: ${error.message}`
+          )
+        }
       }
     }
```

Comment:
```suggestion
  if (accessToken) return

  try {
    // Sets the promise so everyone else will wait - then get the value
    currentlyRequestingPromise = refreshTokenIfNeeded(requestRefresh)
    accessToken = await currentlyRequestingPromise
  } catch (error: unknown) {
    throw new Error(
      `Unable to refresh access token for request due to token refresh error: ${error.message}`
    )
  } finally {
    // Reset the promise
    currentlyRequestingPromise = undefined
  }
```

This is what I mean

Reply:
It really won't work.  This function returns a `requestConfig` with the proper headers set.  Check out the lines below - those lines have to run once you have a valid `accessToken`.